### ISSUE TITLE
add_noise should take key as KeyArrayLike, not seed as int

### ISSUE THREAD

**Issue by Tomas542** [google-deepmind/optax#1137]
I think, that seed in [add_noise ](https://github.com/google-deepmind/optax/blob/5fc395b5e49602e3c0027b976d1368755b20cd0b/optax/_src/transform.py#L693) should be replaced with key to follow `jax.random` and `flax.nnx.Modules`, where keys are used instead of seeds. It doesn't seem hard to do, like replace seed in [args](https://github.com/google-deepmind/optax/blob/5fc395b5e49602e3c0027b976d1368755b20cd0b/optax/_src/transform.py#L693), in [docstring](https://github.com/google-deepmind/optax/blob/5fc395b5e49602e3c0027b976d1368755b20cd0b/optax/_src/transform.py#L703),  and then in [AddNoiseState init](https://github.com/google-deepmind/optax/blob/5fc395b5e49602e3c0027b976d1368755b20cd0b/optax/_src/transform.py#L712) (AddNoiseState by itself doesn't need to change as far as I can see).

**Comment by Tomas542** [comment]
also key init in [AddNoiseState ](https://github.com/google-deepmind/optax/blob/5fc395b5e49602e3c0027b976d1368755b20cd0b/optax/_src/transform.py#L712)should be changed with new `jax.random.key` in favor of [JEP](https://jax.readthedocs.io/en/latest/jep/9263-typed-keys.html)

**Comment by Tomas542** [comment]
I think I can, but I've never done this before. I should check contrib guide on official cite?

**Comment by rdyro** [comment]
That's right. If you want to purse it, it's a valuable contribution.

You can fork the repository, make a new branch in your fork, add the changes (to both code and comments) and update tests (probably no need to write new tests). If you point your browser to this repo, github should offer to make a pull request from your fork.

Feel free to do it incrementally, you can make the PR almost immediately and any additional commits in the branch in your fork will automatically show up in the PR.

I can review your code once you make the PR

**Comment by Tomas542** [comment]
Done! But currently I have no option to run `test.sh` on my local machine. Can I create PR? Or better to set virtual machine with Linux for test?

**Comment by Tomas542** [comment]
Also I additionally changed `noisy_sgd`, cause it's uses `add_noise` and added (duplicated) example of use from `noisy_sgd` to `add_noise`

**Comment by rdyro** [comment]
Cool! Let's continue this discussion on the PR, can you make sure you create it for deepmind/optax? I can't see it at the moment

**Comment by Tomas542** [comment]
here https://github.com/google-deepmind/optax/pull/1138

**Comment by Tomas542** [comment]
close due to merge #1240
